browser-sign-in/OktaBrowserSignIn/AppDelegate.swift

```diff
@@ -15,28 +15,89 @@
  */
 
 import UIKit
-import OktaOidc
+import WebAuthenticationUI
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
-    
     var window: UIWindow?
-
+    var configForUITests: [String: String]? {
+        let env = ProcessInfo.processInfo.environment
+        guard let oktaURL = env["OKTA_URL"], oktaURL.count > 0,
+              let clientID = env["CLIENT_ID"],
+              let redirectURI = env["REDIRECT_URI"],
+              let logoutRedirectURI = env["LOGOUT_REDIRECT_URI"]
+        else {
+            return nil
+        }
+        
+        let configuration = [
+            "issuer": "\(oktaURL)/oauth2/default",
+            "clientId": clientID,
+            "redirectUri": redirectURI,
+            "logoutRedirectUri": logoutRedirectURI,
+            "scopes": "openid profile offline_access"
+        ]
+        return configuration
+    }
+    
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+        let window = UIWindow(frame: UIScreen.main.bounds)
+        window.makeKeyAndVisible()
+        self.window = window
+        
+        if ProcessInfo.processInfo.arguments.contains("--reset-keychain") {
+            try? Keychain.Search().delete()
+        }
+        
+        NotificationCenter.default.addObserver(forName: .defaultCredentialChanged,
+                                               object: nil,
+                                               queue: .main) { notification in
+            self.setRootViewController()
+        }
+        
+        self.setRootViewController()
         return true
     }
     
-    func application(_ app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey : Any]) -> Bool {
-        return false
+    func setRootViewController() {
+        let storyboard = UIStoryboard(name: "Main", bundle: nil)
+        guard Credential.default == nil else {
+            let profileViewController = storyboard.instantiateViewController(withIdentifier: "Profile")
+            self.window?.rootViewController = profileViewController
+            return
+        }
+        
+        let welcomeViewController = storyboard.instantiateViewController(withIdentifier: "SignIn")
+        //Setup for UI Tests
+        if let configForUITests = configForUITests {
+            let _ =  WebAuthentication(
+                issuer: URL(string: configForUITests["issuer"]!)!,
+                clientId: configForUITests["clientId"]!,
+                scopes: configForUITests["scopes"]!,
+                redirectUri: URL(string: configForUITests["redirectUri"]!)!,
+                logoutRedirectUri: URL(string: configForUITests["logoutRedirectUri"]!)!,
```

Comment:
Lets avoid force-unwrapping these values, instead moving them to an `if let` conditional.